plugins/modules/ome_network_vlan_info.py

```diff
@@ -0,0 +1,248 @@
+        }
+      ]
+    }
+  }
+'''
+
+import json
+from ssl import SSLError
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.dellemc.openmanage.plugins.module_utils.ome import RestOME
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.module_utils.urls import ConnectionError, SSLValidationError
+
+# Base URI to fetch all logical networks information
+NETWORK_VLAN_BASE_URI = "NetworkConfigurationService/Networks"
+NETWORK_TYPE_BASE_URI = "NetworkConfigurationService/NetworkTypes"
+QOS_TYPE_BASE_URI = "NetworkConfigurationService/QosTypes"
+
+# Module Success Message
+MODULE_SUCCESS_MESSAGE = "Successfully retrieved the network VLAN information."
+
+# Module Failure Messages
+MODULE_FAILURE_MESSAGE = "Failed to retrieve the network VLAN information."
+NETWORK_VLAN_NAME_NOT_FOUND = "Provided network VLAN with name - '{0}' does not exist."
+
+
+def clean_data(data):
+    """
+    data: A dictionary.
+    return: A data dictionary after removing items that are not required for end user.
+    """
+    for k in ['@odata.id', '@odata.type', '@odata.context', '@odata.count']:
+        data.pop(k, None)
+    return data
+
+
+def get_network_type_and_qos_type_information(rest_obj, network_vlan):
+    """
+    rest_obj: Object containing information about connection to device.
+    network_vlan: A dictionary containing information of network VLAN.
+    return: updated dictionary with additional info for "Type" and "QosType" keys.
+    """
+    network_vlan = clean_data(network_vlan)
+    network_types_uri = "{0}({1})".format(NETWORK_TYPE_BASE_URI, network_vlan.get("Type"))
+    resp = rest_obj.invoke_request('GET', network_types_uri)
+    if resp.status_code == 200:
+        network_vlan['Type'] = clean_data(resp.json_data)
+        qos_type_uri = "{0}({1})".format(QOS_TYPE_BASE_URI, network_vlan.get("Type").get("QosType"))
+        resp = rest_obj.invoke_request('GET', qos_type_uri)
+        network_vlan['Type']['QosType'] = clean_data(resp.json_data) if resp.status_code == 200 else \
+            network_vlan['Type']['QosType']
+    return network_vlan
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec={
+            "hostname": {"required": True, "type": 'str'},
+            "username": {"required": True, "type": 'str'},
+            "password": {"required": True, "type": 'str', "no_log": True},
+            "port": {"required": False, "default": 443, "type": 'int'},
+            "id": {"required": False, "type": 'int'},
+            "name": {"required": False, "type": 'str'}
+        },
+        mutually_exclusive=[["id", "name"]],
+        supports_check_mode=False)
+    try:
+        with RestOME(module.params, req_session=True) as rest_obj:
+            # Form URI to fetch network VLAN information
+            network_vlan_uri = "{0}({1})".format(NETWORK_VLAN_BASE_URI, module.params.get("id")) if module.params.get(

```

Comment:
Add pagination supports to retrieve all available VLAN network information.

Reply:
@Sajna-Shetty Incorporated pagination with safe limit of 9999.